### ISSUE TITLE
[SOLVED] ERROR: Playlist Player: skipping unplayable item

### ISSUE THREAD

**Issue by DavidHenryThoreau** [lekma/plugin.video.invidious#56]
Some video can't be played with error : ERROR: Playlist Player: skipping unplayable item: 0, path

`2021-02-18 18:48:22.401 T:140194040772160  NOTICE: [plugin.video.invidious] request.url: https://invidious.snopyta.org/api/v1/videos/IqOedGIHonY?region=FR
2021-02-18 18:48:35.174 T:140195577150144   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.invidious/?action=video&videoId=IqOedGIHonY]`

**Comment by lekma** [comment]
it seems to be a bug with snopyta (and other instances), sometimes they include an error even if the result of the request is (kinda) valid.

compare:

- https://invidious.snopyta.org/api/v1/videos/IqOedGIHonY?region=FR (error)
- https://invidious.fdn.fr/api/v1/videos/IqOedGIHonY?region=FR (error)
- https://invidious.zapashcanon.fr/api/v1/videos/IqOedGIHonY?region=FR (no error)

there is no obvious way to cleanly and consistently identify these case.

besides that, some instances return a wonky playlist (often based on their geographical location), in that specific case all public instances seem to bork the playlist (only one private instance I have access to returned a playable playlist).

afaict your options are:
- find a better instance
- use the "Play from YouTube" context menu (it was implemented for situations just like this one)

please report if the video plays for you with the YouTube option (it does for me)

thx for your help

p.s.: if you still want to use snopyta, may I suggest you up the timeout to at least 1 min, their connection is mediocre at best.

**Comment by DavidHenryThoreau** [comment]
I'v tried, right click and play on YouTube :

`2021-02-19 16:26:49.581 T:1937387536  NOTICE: VideoPlayer: finished waiting
2021-02-19 16:27:41.922 T:1725289344  NOTICE: [plugin.video.invidious] request.url: https://invidious.zapashcanon.fr/api/v1/videos/IqOedGIHonY?region=FR
2021-02-19 16:27:56.729 T:1626739584  NOTICE: [plugin.video.invidious] video.url: http://127.0.0.1:44335/manifest?videoId=IqOedGIHonY
2021-02-19 16:27:57.797 T:1937387536  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.invidious/?action=video&youtube=true&videoId=IqOedGIHonY
2021-02-19 16:27:57.824 T:1799328640  NOTICE: Creating InputStream
2021-02-19 16:27:58.299 T:1725289344  NOTICE: [plugin.video.invidious] youtube.url: https://www.youtube.com/watch?hl=fr&v=IqOedGIHonY
2021-02-19 16:28:00.931 T:1725289344  NOTICE: [plugin.video.invidious] youtube.url: https://www.youtube.com/s/player/1c732901/player_ias.vflset/fr_FR/base.js
2021-02-19 16:28:18.315 T:1799328640   ERROR: CCurlFile::FillBuffer - Failed: Timeout was reached(28)
2021-02-19 16:28:18.316 T:1799328640   ERROR: CCurlFile::Open failed with code 0 for http://127.0.0.1:44335/manifest?videoId=IqOedGIHonY:
2021-02-19 16:28:18.316 T:1799328640   ERROR: AddOnLog: InputStream Adaptive: Cannot download http://127.0.0.1:44335/manifest?videoId=IqOedGIHonY
2021-02-19 16:28:18.317 T:1799328640   ERROR: AddOnLog: InputStream Adaptive: Could not open / parse mpdURL (http://127.0.0.1:44335/manifest?videoId=IqOedGIHonY)
2021-02-19 16:28:18.320 T:1799328640   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.invidious/?action=video&youtube=true&videoId=IqOedGIHonY]
2021-02-19 16:28:18.325 T:1799328640  NOTICE: CVideoPlayer::OnExit()
2021-02-19 16:28:18.552 T:1937387536  NOTICE: CVideoPlayer::CloseFile()
2021-02-19 16:28:18.554 T:1937387536  NOTICE: VideoPlayer: waiting for threads to exit
2021-02-19 16:28:18.554 T:1937387536  NOTICE: VideoPlayer: finished waiting
2021-02-19 16:28:20.474 T:1725289344  NOTICE: [plugin.video.invidious] "GET /manifest?videoId=IqOedGIHonY HTTP/1.1" 200 -
2021-02-19 16:28:20.520 T:1725289344   ERROR: [plugin.video.invidious] httpd: error processing request: [500 - Internal Server Error]
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/script.module.iapc/lib/iapc/http_server.py", line 94, in _execute
                                                return self.server._execute(self)
                                              File "/storage/.kodi/addons/script.module.iapc/lib/iapc/http_server.py", line 162, in _execute
                                                return request.process_content(*result)
                                              File "/storage/.kodi/addons/script.module.iapc/lib/iapc/http_server.py", line 83, in process_content
                                                self.send_response(code)
                                              File "/usr/lib/python2.7/BaseHTTPServer.py", line 406, in send_response
                                              File "/usr/lib/python2.7/BaseHTTPServer.py", line 412, in send_header
                                              File "/usr/lib/python2.7/socket.py", line 328, in write
                                              File "/usr/lib/python2.7/socket.py", line 307, in flush
                                            error: [Errno 32] Broken pipe
2021-02-19 16:28:20.525 T:1725289344  NOTICE: [plugin.video.invidious] "GET /manifest?videoId=IqOedGIHonY HTTP/1.1" 500 -
2021-02-19 16:28:20.534 T:1725289344   ERROR: Traceback (most recent call last):
2021-02-19 16:28:20.535 T:1725289344   ERROR:   File "/usr/lib/python2.7/SocketServer.py", line 293, in _handle_request_noblock
2021-02-19 16:28:20.536 T:1725289344   ERROR:   File "/usr/lib/python2.7/SocketServer.py", line 321, in process_request
2021-02-19 16:28:20.538 T:1725289344   ERROR:   File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
2021-02-19 16:28:20.539 T:1725289344   ERROR:   File "/usr/lib/python2.7/SocketServer.py", line 657, in __init__
2021-02-19 16:28:20.540 T:1725289344   ERROR:   File "/usr/lib/python2.7/SocketServer.py", line 716, in finish
2021-02-19 16:28:20.541 T:1725289344   ERROR:   File "/usr/lib/python2.7/socket.py", line 283, in close
2021-02-19 16:28:20.542 T:1725289344   ERROR:   File "/usr/lib/python2.7/socket.py", line 307, in flush
2021-02-19 16:28:20.544 T:1725289344   ERROR: error: [Errno 32] Broken pipe`

**Comment by lekma** [comment]
> 2021-02-19 16:28:18.315 T:1799328640 ERROR: CCurlFile::FillBuffer - Failed: Timeout was reached(28) 

bad connection to youtube?

can you also make sure that the service is running, you should have smthing like the following in your log:

2021-02-19 16:37:22.553 T:139797259597568  NOTICE: [plugin.video.invidious] starting service...
2021-02-19 16:37:22.555 T:139797259597568  NOTICE: [plugin.video.invidious] service.url: 'https://invidious.zapashcanon.fr/api/v1/'
2021-02-19 16:37:22.555 T:139797259597568  NOTICE: [plugin.video.invidious] service.timeout: (30.05, 30.0)
2021-02-19 16:37:22.555 T:139797259597568  NOTICE: [plugin.video.invidious] httpd: server started on: ('127.0.0.1', 58579)

it might be that the local http server cannot connect to youtube or is not running

**Comment by DavidHenryThoreau** [comment]
Yeah service is started :

I've tried right click and play on YouTube with another video, it ok : 

`2021-02-19 17:19:28.565 T:1724531584  NOTICE: [plugin.video.invidious] starting service...
2021-02-19 17:19:28.866 T:1724531584  NOTICE: [plugin.video.invidious] service.url: 'https://invidious.zapashcanon.fr/api/v1/'
2021-02-19 17:19:28.869 T:1724531584  NOTICE: [plugin.video.invidious] service.timeout: (60.049999999999997, 60.0)
2021-02-19 17:19:29.005 T:1724531584  NOTICE: [plugin.video.invidious] httpd: server started on: ('127.0.0.1', 34561)
2021-02-19 17:20:35.748 T:1724531584  NOTICE: [plugin.video.invidious] request.url: https://invidious.zapashcanon.fr/api/v1/channels/UCNKMpnM_Yvf6E-Hhf9btYqA/videos?region=FR&page=1
2021-02-19 17:20:43.563 T:1724531584  NOTICE: [plugin.video.invidious] request.url: https://invidious.zapashcanon.fr/api/v1/channels/UCNKMpnM_Yvf6E-Hhf9btYqA?region=FR
2021-02-19 17:21:30.431 T:1724531584  NOTICE: [plugin.video.invidious] request.url: https://invidious.zapashcanon.fr/api/v1/videos/1aygs4EPbAU?region=FR
2021-02-19 17:21:34.339 T:1625375616  NOTICE: [plugin.video.invidious] video.url: http://127.0.0.1:34561/manifest?videoId=1aygs4EPbAU
2021-02-19 17:21:35.466 T:1937219600  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.invidious/?action=video&youtube=true&videoId=1aygs4EPbAU
2021-02-19 17:21:35.496 T:1799328640  NOTICE: Creating InputStream
2021-02-19 17:21:37.328 T:1724531584  NOTICE: [plugin.video.invidious] youtube.url: https://www.youtube.com/watch?hl=fr&v=1aygs4EPbAU
2021-02-19 17:21:40.664 T:1724531584  NOTICE: [plugin.video.invidious] "GET /manifest?videoId=1aygs4EPbAU HTTP/1.1" 200 -
2021-02-19 17:21:40.810 T:1799328640  NOTICE: Creating Demuxer
2021-02-19 17:21:40.834 T:1799328640  NOTICE: Opening stream: 1001 source: 256
2021-02-19 17:21:42.221 T:1799328640  NOTICE: Creating video thread
2021-02-19 17:21:42.224 T:1799328640  NOTICE: Opening stream: 1003 source: 256
2021-02-19 17:21:42.513 T:1799328640  NOTICE: Creating audio thread`

**Comment by lekma** [comment]
do you mean to say that only this video fails to play from youtube?

something else you can try to debug, while kodi and the addon are running, connect to the local http server with a browser on the same machine using the video.url you will find in the log, example from my log:

2021-02-19 17:36:12.798 T:140488833226496  NOTICE: [plugin.video.invidious] video.url: http://127.0.0.1:33091/manifest?videoId=IqOedGIHonY

it should return a manifest playlist.
if you don't have a browser on the same machine but have shell access you could use wget:

`wget -q -O - http://127.0.0.1:33091/manifest?videoId=IqOedGIHonY`

otherwise I must admit I'm at a loss: if you're correct that only this video doesn't play, I must say I cannot think of anything explaining the situation. it plays fine from here.

> 2021-02-19 17:19:28.869 T:1724531584 NOTICE: [plugin.video.invidious] service.timeout: (60.049999999999997, 60.0)

this is strange, what version of kodi/python/plugin are you running?

**Comment by DavidHenryThoreau** [comment]
> do you mean to say that only this video fails to play from youtube?
> 

At the moment the only video wich can't be played by native Invidious plugin and right click play on YouTube is (it may have others but i don't remember) :
IqOedGIHonY

> something else you can try to debug, while kodi and the addon are running, connect to the local http server with a browser on the same machine using the video.url you will find in the log, example from my log:
> 
> 2021-02-19 17:36:12.798 T:140488833226496 NOTICE: [plugin.video.invidious] video.url: http://127.0.0.1:33091/manifest?videoId=IqOedGIHonY
> 
> it should return a manifest playlist.
> if you don't have a browser on the same machine but have shell access you could use wget:
> 
> `wget -q -O - http://127.0.0.1:33091/manifest?videoId=IqOedGIHonY`
> 

The wget return the plylist yes.

httpd state in log 

`2021-02-19 19:06:22.130 T:1727206272  NOTICE: [plugin.video.invidious] httpd: server started on: ('127.0.0.1', 45867)
2021-02-19 19:08:00.348 T:1727206272   ERROR: [plugin.video.invidious] httpd: error processing request: [500 - Internal Server Error]
2021-02-19 19:08:49.723 T:1727206272   ERROR: [plugin.video.invidious] httpd: error processing request: [500 - Internal Server Error]`


> otherwise I must admit I'm at a loss: if you're correct that only this video doesn't play, I must say I cannot think of anything explaining the situation. it plays fine from here.
> 
> > 2021-02-19 17:19:28.869 T:1724531584 NOTICE: [plugin.video.invidious] service.timeout: (60.049999999999997, 60.0)
> 
> this is strange, what version of kodi/python/plugin are you running?

I'm running Raspberry Pi 1 with LibreELEC 9.2.6 and Kodi 18.9 
My old Raspberry may can't decodes some videos

**Comment by lekma** [comment]
what is the python version?

**Comment by lekma** [comment]
can you try: ping localhost on the rpi?

**Comment by DavidHenryThoreau** [comment]
Python 2.7.16

`ping localhost
PING localhost (127.0.0.1): 56 data bytes
64 bytes from 127.0.0.1: seq=0 ttl=64 time=0.996 ms`

**Comment by MrRoy** [comment]
I'm having the same error as the OP:
`2021-05-25 13:06:06.135 T:20548    INFO <general>: initializing python engine.`
`2021-05-25 13:06:06.581 T:5752     INFO <general>: [plugin.video.invidious] <service.session> request: http://invidious/api/v1/videos/MD_lxVIwO2Y?region=US`
`2021-05-25 13:06:09.051 T:23616   ERROR <general>: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.invidious/?action=video&videoId=MD_lxVIwO2Y]`
`2021-05-25 13:06:09.650 T:20548    INFO <general>: CPythonInvoker(4, C:\Users\jroy\AppData\Roaming\Kodi\addons\plugin.video.invidious\lib\plugin.py): script successfully run
`

I'm using my own instance so I'm hoping there's something I can fix on it if it is sending borked playlists. Needless to say, it works fine when I'm using it from my browser.
I tried downloading the manifest from `http://127.0.0.1:33091/manifest?videoId=` and it worked, but when I tried to access the googlevideo URLs through my browser, I got 403 errors, could this be the problem, or is it normal to get 403s when accessing those URLs from a web browser?

The "play from YouTube" function also does not work, but there is no error in the log, I only get "video unavailable" message, but I find it weird that the log tries to query my invidious instance anyway?:
`2021-05-25 13:05:11.795 T:23616 WARNING <general>: CGUIWindowManager - CGUIWindowManager::HandleAction - ignoring action 107, because topmost modal dialog closing animation is running`
`2021-05-25 13:05:15.942 T:6324     INFO <general>: Skipped 36 duplicate messages..`
`2021-05-25 13:05:15.942 T:6324     INFO <general>: initializing python engine.`
`2021-05-25 13:05:15.968 T:20436    INFO <general>: Python interpreter stopped`
`2021-05-25 13:05:16.665 T:6324     INFO <general>: Skipped 5 duplicate messages..`
`2021-05-25 13:05:16.665 T:6324     INFO <general>: CPythonInvoker(3, C:\Users\jroy\AppData\Roaming\Kodi\addons\plugin.video.invidious\lib\script.py): script successfully run`
`2021-05-25 13:05:16.674 T:20548    INFO <general>: initializing python engine.`
`2021-05-25 13:05:16.688 T:6324     INFO <general>: Python interpreter stopped`
`2021-05-25 13:05:17.273 T:5752     INFO <general>: [plugin.video.invidious] <service.session> request: http://invidious/api/v1/videos/0WVDKZJkGlY?region=US`
`2021-05-25 13:05:21.286 T:8284     INFO <general>: CActiveAESink::OpenSink - initialize sink`
`2021-05-25 13:05:21.955 T:20548    INFO <general>: CPythonInvoker(4, C:\Users\jroy\AppData\Roaming\Kodi\addons\plugin.video.invidious\lib\plugin.py): script successfully run`
